chain/vm/runtime.go

```diff
@@ -295,15 +295,27 @@ func (rt *Runtime) ValidateImmediateCallerIs(as ...address.Address) {
 			return
 		}
 	}
-	rt.Abortf(exitcode.SysErrForbidden, "caller %s is not one of %s", rt.Message().Caller(), as)
+	abortf(exitcode.SysErrForbidden, "caller %s is not one of %s", rt.Message().Caller(), as)
 }
 
 func (rt *Runtime) Context() context.Context {
 	return rt.ctx
 }
 
+// Abortf should be called by ACTORS to abort execution with the given non-system
+// and non-negative exit code.
 func (rt *Runtime) Abortf(code exitcode.ExitCode, msg string, args ...interface{}) {
 	log.Warnf("Abortf: " + fmt.Sprintf(msg, args...))
+	if code < exitcode.FirstActorErrorCode {
+		panic(aerrors.NewfSkip(2, exitcode.SysErrorIllegalActor, "Actor used illegal exit code: %s", code))
+	}
+	panic(aerrors.NewfSkip(2, code, msg, args...))
+}
+
+// abortf should be called by the runtime/vm to abort execution with ANY exit
+// code, including system codes.
+func abortf(code exitcode.ExitCode, msg string, args ...interface{}) {
```

Comment:
Can we call this `vmabortf` to make it evident that it's the VM that's aborting?